not robust against some malformed messages

try

```
deploys.test.myservice4:100|t
```

"t" is wrong, it should be "ms" if I'm trying to test timers. But it probably shouldn't crash statsdaemon:

```
panic: interface conversion: interface is uint64, not int64

goroutine 16 [running]:
runtime.panic(0x52e740, 0xc20802e100)
    /usr/local/Cellar/go/1.3/libexec/src/pkg/runtime/panic.c:279 +0xf5
main.packetHandler(0xc20802e0c0)
    /Users/jehiah/projects/gopath/src/github.com/bitly/statsdaemon/statsdaemon.go:129 +0x603
main.monitor()
    /Users/jehiah/projects/gopath/src/github.com/bitly/statsdaemon/statsdaemon.go:101 +0x488
```

--- a/statsdaemon.go
+++ b/statsdaemon.go
@@ -123,7 +123,7 @@ func packetHandler(s *Packet) {
 		timers[s.Bucket] = append(timers[s.Bucket], s.Value.(uint64))
 	} else if s.Modifier == "g" {
 		gauges[s.Bucket] = s.Value.(uint64)
-	} else {
+	} else if s.Modifier == "c" {
 		_, ok := counters[s.Bucket]
 		if !ok {
 			counters[s.Bucket] = 0

--- a/statsdaemon_test.go
+++ b/statsdaemon_test.go
@@ -116,6 +116,15 @@ func TestPacketParse(t *testing.T) {
 	assert.Equal(t, len(packets), 1)
 }
 
+func TestMalformedDataHandling(t *testing.T) {
+
+	// reported as issue #45
+	d := []byte("deploys.test.myservice4:100|t")
+	packets := parseMessage(d)
+	packetHandler(packets[0])
+
+}
+
 func TestReceiveCounterPacketHandling(t *testing.T) {
 	counters = make(map[string]int64)
 	*receiveCounter = "countme"

